docs/reference/sql/functions/date-time.asciidoc

```diff
@@ -721,6 +721,53 @@ include-tagged::{sql-specs}/docs/docs.csv-spec[truncateIntervalHour]
 include-tagged::{sql-specs}/docs/docs.csv-spec[truncateIntervalDay]
 --------------------------------------------------
 
+[[sql-functions-datetime-format]]
+==== `FORMAT`
+
+.Synopsis:
+[source, sql]
+--------------------------------------------------
+FORMAT(
+    date_exp/datetime_exp/time_exp, <1>
+    string_exp) <2>
+--------------------------------------------------
+
+*Input*:
+
+<1> date/datetime/time expression
+<2> format pattern
+
+*Output*: string
+
+*Description*: Returns the date/datetime/time as a string using the format specified in the 2nd argument. The formatting
+pattern used is the one from
+https://docs.microsoft.com/en-us/sql/t-sql/functions/format-transact-sql?view=sql-server-ver15#ExampleD[Microsoft SQL Server Format Specification].
+If any of the two arguments is `null` or the pattern is an empty string `null` is returned.
+
+[NOTE]
+If the 1st argument is of type `time`, then pattern specified by the 2nd argument cannot contain date related units
+(e.g. 'dd', 'MM', 'YYYY', etc.). If it contains such units an error is returned.
+
+[NOTE]
+Format specifier `F` will be working similar to format specifier `f`.
+It represents the most significant digit of the seconds fraction and all trailing `0` s will be printed to meet the number of `F` s in the pattern,
+e.g.: for a time part `10:20:30.1234` and pattern `HH:mm:ss.FFFFFF`, the output string of the function would be: `10:20:30.123400`.
```

Comment:
```suggestion
Diverging from the reference documentation, format specifier `F` behaves identical to format specifier `f`.
```

IMO the docs for `f` should explain sufficiently `F`'s behavior, we shouldn't need an extra example.

But stepping back, if `f`==`F`, why do we need `F` at all? @matriv, do we want to plug into some SQL Server's FORMAT usages? Otherwise I guess it could be dropped altogether.

Reply:
Since we say that we support the patterns of SQL Server's `FORMAT` and `F` is the only exception we don't support exactly as in SQL Server (we still print trailing 0s), I thought that we should still support it but explain that has the exact same effect as `f`.

Reply:
IMO since we need to make a note about it anyways and it brings no new functionality, it'd be cleaner to just drop it and note that instead. But not feeling strong about, it can remain as is too.

Reply:
I don't have strong opinion but a slight preference on keeping it and document it, since the functionality of `FORMAT` is to ease the life of a user acquainted with SQL Server.
On the other hand, if the user doesn't read the docs could just use `F` and face confusing results.

If we choose to drop it, we'd have to return an error message in resolveType() and in the Processor when the `F` is contained in the pattern.

@bpintea, @binu-r What's your final call?

Reply:
Sure, this also makes sense, let's keep it then.

Reply:
I also prefer in keeping it with notes.

Reply:
I find the existing formulation actually a bit confusing: since multiple `F`s are accepted this thew me off a bit initially: `It represents the most significant digit of the seconds fraction and all trailing `0` s will be printed to meet the number of `F` s in the pattern`: 
* just the first `F` corresponds to `the most significant digit`;
* there are multiple fractions a second has;
* minor, the space in "`F` s" and "`0` s" can be dropped too;
* the `and all the trailing 0...` doesn't logically follow one `most significant digit`.

So yes, although I find the original Microsoft documentation telling enough, it doesn't have to be necessarily my proposal, but it should be changed nevertheless to make it clearer.

Otherwise LGTM, thanks.

Reply:
@bpintea updated documentation. Thanks.